### ISSUE TITLE
Compilation failing with missing uw_Basis_urlifyChar_w

### ISSUE THREAD

**Issue by fabriceleal** [urweb/urweb#159]
I was changing my utf8.ur test to ensure that the code i want is _really_ called on the server side (turns out, it was all being called on the client side) and I stumbled upon an compilation error that I think you can reproduce by forking [this branch](https://github.com/fabriceleal/urweb/tree/testing-fix-codepoint-js) and trying to build-run the test by running `./tests/driver.sh utf8`

**Comment by fabriceleal** [comment]
The error I get is:

```
gcc  -pthread -Wimplicit -Werror -Wno-unused-value -I /usr/local/include/urweb   -c "/tmp/webapp.c" -o "/tmp/webapp.o" -g
/tmp/webapp.c: In function ‘uw_handle’:
/tmp/webapp.c:11356:28: error: implicit declaration of function ‘uw_Basis_urlifyChar_w’; did you mean ‘uw_Basis_sqlifyChar’? [-Werror=implicit-function-declaration]
                            uw_Basis_urlifyChar_w(ctx, it1);
                            ^~~~~~~~~~~~~~~~~~~~~
                            uw_Basis_sqlifyChar
cc1: all warnings being treated as errors

```

I'm fairly sure the issue is here:
https://github.com/fabriceleal/urweb/blob/testing-fix-codepoint-js/tests/utf8.ur#L86
TLDR: I'm trying to bring `char` from the server side via `rpc`

**Comment by achlipala** [comment]
Interesting: somehow up to now everyone managed to avoid any code that needs to encode `char` values in URLs in server-side code!  The C functions for such encoding are not written, though they are almost certainly subsets of the functionality for `string` encoding.

Do you have any interest in tackling the addition of this C code?  It is indeed related to your internationalization changes, and I think you already wrote the relevant hard/tedious bits of code.  Ideally, `src/mono_opt.sml` would also be updated to do compile-time encoding of constants, just as is done for other datatypes now.

**Comment by fabriceleal** [comment]
Yeah, I'll add it and will try to check how `mono_opt.sml` works.
